src/microbe_stage/Microbe.Interior.cs

```diff
@@ -327,6 +337,45 @@ public void QueueEmitToxin(Compound toxinCompound)
         queuedToxinToEmit = toxinCompound;
     }
 
+    /// <summary>
+    ///   Tries to secrete slime if possible
+    /// </summary>
+    public void SecreteSlime(float delta)
+    {
+        if (SlimeJets.Count < 1)
+            return;
+
+        if (slimeSecretionCooldown > 0)
+            return;
+
+        var totalSlimeSecreted = 0.0f;
+
+        // Reset the jet factor before calculating to prevent accidental build up of speed
+        slimeJetFactor = Vector3.Zero;
+
+        // Secrete slime and gather overall force from directions determined by equipped jets
+        foreach (var jet in SlimeJets)
+        {
+            var slimeSecreted = EjectCompound(mucilage, Constants.COMPOUNDS_TO_VENT_PER_SECOND * delta,
+                -jet.GetDirection(), 2);
+            slimeJetFactor += slimeSecreted * jet.GetDirection();
```

Comment:
As `jet.GetDirection()` runs a few lines of computations, it may be relevant to store it somewhere.